src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/ChangeApprenticeshipSteps.cs

```diff
@@ -122,34 +152,79 @@ public void ThenTheCommitmentStatementExistsInDatabase()
+                {
+                    _request.EmployerAccountLegalEntityId,
+                    _request.EmployerName,
+                    _request.TrainingProviderId,
+                    _request.TrainingProviderName,
+                    Course = new
+                    {
+                        Name = _request.CourseName,
+                        Level = _request.CourseLevel,
+                        Option = _request.CourseOption,
+                        _request.PlannedStartDate,
+                        _request.PlannedEndDate,
+                    },
+                }
+            });
+        }
+
         [Then("the new commitment statement has same commitments apprenticeship Id")]
         public void ThenTheCommitmentStatementHasSameCommitmentsApprenticeshipId()
         {
-            var cs = _context.DbContext.CommitmentStatements.ToList();
-
             _context.DbContext.CommitmentStatements.Should().ContainEquivalentOf(new
             {
-                CommitmentsApprovedOn = _request.ApprovedOn,
+                CommitmentsApprovedOn = _request.CommitmentsApprovedOn,
                 CommitmentsApprenticeshipId = _commitmentStatement.CommitmentsApprenticeshipId
             });
         }
 
         [Then(@"the new commitment statement has a new commitments apprenticeship Id")]
         public void ThenTheNewCommitmentStatementHasANewCommitmentsApprenticeshipId()
         {
-            var cs = _context.DbContext.CommitmentStatements.ToList();
-
             _context.DbContext.CommitmentStatements.Should().ContainEquivalentOf(new
             {
-                CommitmentsApprovedOn = _request.ApprovedOn,
+                CommitmentsApprovedOn = _request.CommitmentsApprovedOn,
                 CommitmentsApprenticeshipId = _newApprenticeshipId
             });
         }
 
+        [Then(@"registration commitments apprenticeship are updated correctly")]
+        public void ThenRegistrationCommitmentsApprenticeshipAreUpdatedCorrectly()
+        {
+            _context.DbContext.Registrations.Should().ContainEquivalentOf(new
+            {
+                CommitmentsApprovedOn = _request.CommitmentsApprovedOn,
+                CommitmentsApprenticeshipId = _request.CommitmentsApprenticeshipId
+            });
+        }
+
         [Then("there should be no commitment statements in the database")]
         public void ThenThereShouldBeNoCommitmentStatementsInTheDatabase()
         {
             _context.DbContext.CommitmentStatements.Should().BeEmpty();
         }
+
+        [Then(@"a domain exception is thrown")]
+        public async Task ThenADomainExceptionIsThrown()
+        {
+            var content = await _context.Api.Response.Content.ReadAsStringAsync();
+            var errors = JsonConvert.DeserializeObject<List<ErrorItem>>(content);
+            errors.Count.Should().Be(1);
+            errors[0].PropertyName.Should().BeNull();
+            errors[0].ErrorMessage.Should().NotBeNull();
```

Comment:
If we do chose to keep `BadRequest` instead of `NotFound`, could we test for valid values in the message?  Asserting that these are not null seems to provide only minor benefit.